### PR TITLE
Added MarketComboBoxTestWidget

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

File: Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  layout->setContentsMargins({});
+  auto database = GetDefaultMarketDatabase();
+  auto combo_box = new MarketComboBox(database, this);
+  layout->addWidget(combo_box);
+  combo_box->connect_selected_signal([=] (auto value) {
+    m_selected_signal(QVariant::fromValue<MarketToken>(value));
+  });
+}
+
+connection MarketComboBoxTestWidget::connect_selected_signal(
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp:23]: What will ever call this method?

Comment (spireken) [reply Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp:23]: The ComboBoxTestWidget calls it to connect the selection signal.

Comment (spireken) [reply Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp:23]: It should be marked override, though.

File: Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+MarketComboBoxTestWidget::MarketComboBoxTestWidget(QWidget* parent)
+    : ComboBoxAdapter(parent) {
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins({});
+  auto database = GetDefaultMarketDatabase();
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/MarketComboBoxTestWidget.cpp:15]: Inline the expression directly unless you intend to use the expression multiple times.

**Review by kranar — APPROVED** [review]

**Pull request merged**